libfwupdplugin/fu-device.c

```diff
@@ -2515,6 +2515,10 @@ fu_device_prepare_firmware (FuDevice *self,
  * @error: A #GError
  *
  * Reads firmware from the device by calling a plugin-specific vfunc.
+ * The firmware will not contain any serial numbers or user-configuration values
+ * and can be used to calculate the device checksum.
```

Comment:
Isn't this assuming that the device SPI is partitioned in a such that the configuration specific stuff is in a certain region and can be ignored for a checksum purpose?  It seems like a general statement that can't be guaranteed.

Reply:
Yes, it's assuming that the specific FuDevice subclass knows how to parse the firmware and blank something out at a known offset. Maybe the wording is just funky; how about `The device subclass should try to ensure the firmware does not contain any serial numbers or user-configuration values...`

Reply:
That's better